File: openapi-generator/csharp-dotnet2/src/main/CsharpDotNet2/Org/OpenAPITools/Model/DoorResponseSensors.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace Org.OpenAPITools.Model {

  /// <summary>
  /// 
  /// </summary>
  [DataContract]
  public class DoorResponseSensors {
    /// <summary>
    /// Flag indicating whether the current door is closed or open.
    /// </summary>
    /// <value>Flag indicating whether the current door is closed or open.</value>
    [DataMember(Name="doorClosed", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "doorClosed")]
    public bool? DoorClosed { get; set; }

    /// <summary>
    /// Name of the sensor.
    /// </summary>
    /// <value>Name of the sensor.</value>
    [DataMember(Name="name", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "name")]
    public string Name { get; set; }

    /// <summary>
    /// ID of the sensor.
    /// </summary>
    /// <value>ID of the sensor.</value>
    [DataMember(Name="id", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "id")]
    public long? Id { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class DoorResponseSensors {\n");
      sb.Append("  DoorClosed: ").Append(DoorClosed).Append("\n");
      sb.Append("  Name: ").Append(Name).Append("\n");
      sb.Append("  Id: ").Append(Id).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}

File: openapi-generator/dart/lib/model/door_response_sensors.dart
part of openapi.api;

class DoorResponseSensors {
  /* Flag indicating whether the current door is closed or open. */
  bool doorClosed = null;
  /* Name of the sensor. */
  String name = null;
  /* ID of the sensor. */
  int id = null;
  DoorResponseSensors();

  @override
  String toString() {
    return 'DoorResponseSensors[doorClosed=$doorClosed, name=$name, id=$id, ]';
  }

  DoorResponseSensors.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    if (json['doorClosed'] == null) {
      doorClosed = null;
    } else {
          doorClosed = json['doorClosed'];
    }
    if (json['name'] == null) {
      name = null;
    } else {
          name = json['name'];
    }
    if (json['id'] == null) {
      id = null;
    } else {
          id = json['id'];
    }
  }

  Map<String, dynamic> toJson() {
    return {
      'doorClosed': doorClosed,
      'name': name,
      'id': id
    };
  }

  static List<DoorResponseSensors> listFromJson(List<dynamic> json) {
    return json == null ? new List<DoorResponseSensors>() : json.map((value) => new DoorResponseSensors.fromJson(value)).toList();
  }

  static Map<String, DoorResponseSensors> mapFromJson(Map<String, dynamic> json) {
    var map = new Map<String, DoorResponseSensors>();
    if (json != null && json.length > 0) {
      json.forEach((String key, dynamic value) => map[key] = new DoorResponseSensors.fromJson(value));
    }
    return map;
  }
}

